activesupport/test/core_ext/duration_test.rb

```diff
@@ -220,4 +220,71 @@ def test_comparable
     assert_equal(1, (1.minute <=> 1.second))
     assert_equal(1, (61 <=> 1.minute))
   end
+
+  def test_iso8601_parsing_wrong_patterns_with_raise
+    invalid_patterns = ['', 'P', 'PT', 'P1YT', 'T', 'PW', 'P1Y1W', '~P1Y', '.P1Y', 'P1.5Y0.5M', 'P1.5Y1M', 'P1.5MT10.5S']
+    invalid_patterns.each do |pattern|
+      assert_raise ActiveSupport::Duration::ISO8601Parser::ParsingError, pattern.inspect do
+        ActiveSupport::Duration.parse!(pattern)
+      end
+    end
+  end
+  
+  def test_iso8601_parsing_wrong_patterns_with_nil
+    invalid_patterns = ['', 'P', 'PT', 'P1YT', 'T', 'PW', 'P1Y1W', '~P1Y', '.P1Y', 'P1.5Y0.5M', 'P1.5Y1M', 'P1.5MT10.5S']
+    invalid_patterns.each do |pattern|
+      assert_nil ActiveSupport::Duration.parse(pattern), pattern.inspect
+    end
+  end
+
+  def test_iso8601_output
+    # As (1.year + 1.month + 1.day + 1.hour).iso8601 will output P1Y1M1DT3600S (weird), test for these not included
```

Comment:
I think even if they're weird we should still test them to make sure that the values don't change unexpectedly